### ISSUE TITLE
[Sunshine] Installed version doesn't match running version?

### ISSUE THREAD

**Issue by goncalossilva** [ublue-os/bazzite#1359]
### Describe the bug

It appears the installed version of sunshine doesn't match the running version.

Installed:

```
$ rpm -q sunshine
sunshine-2024.713.205505-1.fc40.x86_64
$ ls -la /usr/bin/ | grep sunshine
lrwxrwxrwx.   2 root root           29 Jul 14 02:53 sunshine -> sunshine-2024.713.205505.copr
-rwxr-xr-x.   1 root root     21846256 Jul  8 02:11 sunshine-2024.713.205505.copr
```

Running:

```
$ sunshine --version
[2024:07:14:03:05:33]: Info: Sunshine version: 2024.629.13903.copr
$ /usr/bin/sunshine --version
[2024:07:14:03:05:47]: Info: Sunshine version: 2024.629.13903.copr
$ /usr/bin/sunshine-2024.713.205505.copr --version
[2024:07:14:03:05:51]: Info: Sunshine version: 2024.629.13903.copr
```

<details><summary>Screenshot of the web interface showing the version</summary>

![image](https://github.com/user-attachments/assets/77c1c143-db98-40e4-99dd-a1c7784800d3)

</details> 

The running version indeed appears to be `2024.629.13903`, as that version has [an issue with AMD GPUs](https://github.com/LizardByte/Sunshine/issues/2788) that's been fixed on the later versions, but I can still reproduce.

It's also worth noting that the previous version, between the latest and the one it appears to be stuck in, suffered from the same problem:

```
$ ls -la /usr/bin/ | grep sunshine
lrwxrwxrwx.   2 root root           29 Jul 12 09:50 sunshine -> sunshine-2024.710.151651.copr
-rwxr-xr-x.   1 root root     21846256 Jul  8 02:11 sunshine-2024.710.151651.copr
$ /usr/bin/sunshine-2024.710.151651.copr --version
[gamepad] -- [xone]
[2024:07:13:02:40:10]: Info: Sunshine version: 2024.629.13903.copr
```

(also notice how the binary size is unchanged)

I reviewed the [build logs](https://download.copr.fedorainfracloud.org/results/matte-schwartz/sunshine/fedora-39-x86_64/07731324-sunshine/) for both versions over on Copr and they look correct.

I suspect I'm missing something fairly obvious. How can this be?

### What did you expect to happen?

I expect the installed version of sunshine to match the runtime version.

### Output of `rpm-ostree status`

```shell
$ rpm-ostree status
State: idle
Deployments:
● ostree-image-signed:docker://ghcr.io/ublue-os/bazzite:stable
                   Digest: sha256:3c0bf8b0e58a578e34d8af30e3da10f547c782d9d3a971502948889388c51883
                  Version: 40.20240713.0 (2024-07-14T00:57:44Z)
          LayeredPackages: sunshine
            LocalPackages: lact-0.5.4-0.x86_64

  ostree-image-signed:docker://ghcr.io/ublue-os/bazzite:stable
                   Digest: sha256:406696eac5e410baa5cd9524922551c5bfa177639cdb506180926d14323d9f76
                  Version: 40.20240712.0 (2024-07-12T04:55:54Z)
          LayeredPackages: sunshine
            LocalPackages: lact-0.5.4-0.x86_64
```

### Hardware

Standard desktop, with a Ryzen 3950X and an AMD 6900XT. Bazzite is a very recent install, with no customizations except installing LACT and Sunshine using `ujust`.

**Comment by goncalossilva** [comment]
This behavior seems specific to Bazzite, or better, my installation of Bazzite.

On a regular Fedora 40 install everything works as expected:

```
$ dnf copr enable matte-schwartz/sunshine 
...
$ dnf install sunshine
<installs 2024.713.205505-1.fc40 + dependencies>
$ sunshine --version
[2024:07:14:22:57:17]: Info: Sunshine version: 2024.713.205505.copr
```

**Comment by SvdB-nonp** [comment]
I coincedently came across this issue and as I was installing sunshine anyway, was checking my versions.

Seeing the outputs, it is working correctly on my end, which is new-ish install (removed a month ago, so some user files may have lingered) with `ujust setup-sunshine`.

```
$ rpm -q sunshine
sunshine-2024.713.205505-1.fc40.x86_64

$ ls -la /usr/bin/ | grep sunshine
lrwxrwxrwx.   2 root root           29 17 jul 12:45 sunshine -> sunshine-2024.713.205505.copr
-rwxr-xr-x.   1 root root     22367680 17 jul 12:54 sunshine-2024.713.205505.copr

$ sunshine --version
[address_family] -- [both]
[output_name] -- [0]
[2024:07:17:12:55:58]: Info: Sunshine version: 2024.713.205505.copr

$ /usr/bin/sunshine --version
[address_family] -- [both]
[output_name] -- [0]
[2024:07:17:12:56:16]: Info: Sunshine version: 2024.713.205505.copr

$ /usr/bin/sunshine-2024.713.205505.copr --version
[address_family] -- [both]
[output_name] -- [0]
[2024:07:17:12:56:30]: Info: Sunshine version: 2024.713.205505.copr
```

And the web interface mentions "Version 2024.713.205505.copr" in the homepage.

**Comment by goncalossilva** [comment]
This issue may stem from my misunderstanding of how Universal Blue. Could it be that layered packages are only updated when the base image is?

Regardless, I found a convoluted way to make it work, by using `rpm-ostree override replace <copr url>` followed by `rpm-ostree apply-live --alow-replacement` and restarting `sunshine`.

Rebooting the system immediately reverts to the previous state.

**Comment by rcarmo** [comment]
I still can't get sunshine to work on current Bazzite. It immediately core demps on startup now, and I have to run a 6 months' old snapshot. I wonder if it's actually being maintained.

**Comment by KyleGospo** [comment]
> I still can't get sunshine to work on current Bazzite. It immediately core demps on startup now, and I have to run a 6 months' old snapshot. I wonder if it's actually being maintained.

Should be working, definitely is on my end. The version that we ship is built on copr from source. Can you provide more details on the crash and your current hardware?

**Comment by rcarmo** [comment]
Same hardware as #1350. I just get an instant core dump, and the user systemd unit reports that but there is no more info.

**Comment by goncalossilva** [comment]
I can't reproduce #1710, but the problem described in the OP persists, assuming it's not [a misunderstanding on my part](https://github.com/ublue-os/bazzite/issues/1359#issuecomment-2234008089).

Currently, I am stuck on `v2024.729.24508`, which is multiple versions behind. The latest is [`v2024.1002.214`](https://github.com/LizardByte/Sunshine/releases/tag/v2024.1002.214) (released today) and the latest available on Copr is [`v2024.911.215654`](https://copr.fedorainfracloud.org/coprs/matte-schwartz/sunshine/packages/) (released a couple of weeks ago). I _can_ get it updated if I follow the steps [I described before](https://github.com/ublue-os/bazzite/issues/1359#issuecomment-2234008089), but as soon as I reboot, it lists `v2024.1002.214`, but it is actually running `v2024.729.24508` per sunshine's webui.

The latest versions are more performant on my particularly setup (Linux + AMD) so I always end up doing this dance.

**Comment by wolfyreload** [comment]
I'm on bazzite-deck:stable-42.20250425

Note that there is a "sunshine" and a "Sunshine" package, which is a little weird. I seem to recall that "sunshine" was the original package that the bazzite team packaged and they've converted it to an empty package and "Sunshine" was the package that upstream packaged. "sunshine" is currently an empty placeholder.

`rpm -q Sunshine` note the capital S

```
Sunshine-2025.423.24313-1.fc42.x86_64
```

`ls -la /usr/bin/ | grep sunshine`

```
lrwxrwxrwx.  1 root root           24 Jan  1  1970 sunshine -> sunshine-v2025.423.24313
-rwxr-xr-x.  1 root root     23767080 Jan  1  1970 sunshine-v2025.423.24313
```

The versions are matching @noelmiller I think you can close this

**Comment by CheckYourFax** [comment]
Yeah that looks okay. I also tested it myself and versions are matching.

**Comment by goncalossilva** [comment]
Thanks, everyone. I can confirm this looks alright on my end too!
